utils/publish-rpm.sh

```diff
@@ -141,34 +150,41 @@ init_cos_utils)
 repo_init)
     # create basic repo directory structure
     # useful when a new repo added
-    func_repo_init /tmp
+    func_repo_init /tmp "$dist"
     ;;
 repo_backup)
-    func_repo_backup "${VAR_COS_BUCKET_REPO}" "centos" "${TAG_DATE}"
+    func_repo_backup "${VAR_COS_BUCKET_REPO}" "$dist" "${TAG_DATE}"
     ;;
 repo_clone)
-    func_repo_clone "${VAR_COS_BUCKET_REPO}" "centos" /tmp/centos
+    func_repo_clone "${VAR_COS_BUCKET_REPO}" "$dist" /tmp/"$dist"
     ;;
 repo_package_sync)
-    VAR_REPO_MAJOR_VER=(7 8)
-    for i in "${VAR_REPO_MAJOR_VER[@]}"; do
-        find "${VAR_RPM_WORKBENCH_DIR}" -type f -name "*el${i}.${ARCH}.rpm" \
-            -exec echo "repo sync for: {}" \; \
-            -exec cp -a {} /tmp/centos/"${i}"/${ARCH} \;
-    done
+    if [[ "$dist" == "centos" ]];then
+        VAR_REPO_MAJOR_VER=(7 8)
+        for i in "${VAR_REPO_MAJOR_VER[@]}"; do
+            find "${VAR_RPM_WORKBENCH_DIR}" -type f -name "*el${i}.${ARCH}.rpm" \
+                -exec echo "repo sync for: {}" \; \
+                -exec cp -a {} /tmp/centos/"${i}"/${ARCH} \;
+        done
+    else #redhat ubi
+        find "${VAR_RPM_WORKBENCH_DIR}" -type f -name "*ubi8.6.${ARCH}.rpm" \
+        -exec echo "repo sync for: {}" \; \
+        -exec cp -a {} /tmp/redhat/ubi8.6/${ARCH} \;
```

Comment:
```suggestion
        -exec cp -a {} /tmp/redhat/8/${ARCH} \;
```

This directory should be consistent with the directory above, otherwise it will fail

Reply:
Made this change